test/bls.test.ts

```diff
@@ -1,22 +1,32 @@
-import { TestBlsInstance } from "../types/truffle-contracts";
+import { TestBlsFactory } from "../types/ethers-contracts/TestBlsFactory";
+import { TestBls } from "../types/ethers-contracts/TestBls";
+import web3 from "web3";
+import { assert } from "chai";
 
-const TestBLS = artifacts.require("TestBLS");
 import * as mcl from "./utils/mcl";
-import { bn, bnToHex, ZERO } from "./utils/mcl";
+import { ethers } from "@nomiclabs/buidler";
 
 describe("BLS", async () => {
-    let bls: TestBlsInstance;
+    let bls: TestBls;
     before(async function() {
         await mcl.init();
-        bls = await TestBLS.new();
+        const accounts = await ethers.getSigners();
+        bls = await new TestBlsFactory(accounts[0]).deploy();
+        await bls.deployed();
     });
     it("hash to point", async function() {
         for (let i = 0; i < 20; i++) {
             const data = web3.utils.randomHex(12);
             let expect = mcl.g1ToHex(mcl.hashToPoint(data));
-            let res = await bls.hashToPoint(data);
-            assert.equal(expect[0], bnToHex(res[0]));
-            assert.equal(expect[1], bnToHex(res[1]));
+            let res = await bls.hashToPoint(ethers.utils.arrayify(data));
```

Comment:
The contract functions now take bigNumberish for integer types. 
But for the "bytes" type, we need to turn the hex into `arrayish` by using `ethers.utils.arrayify`